x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/plan/physical/MergeExec.java

```diff
@@ -0,0 +1,75 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+package org.elasticsearch.xpack.esql.plan.physical;
+
+import org.elasticsearch.common.io.stream.StreamOutput;
+import org.elasticsearch.xpack.esql.core.expression.Attribute;
+import org.elasticsearch.xpack.esql.core.tree.NodeInfo;
+import org.elasticsearch.xpack.esql.core.tree.Source;
+import org.elasticsearch.xpack.esql.plan.logical.local.LocalSupplier;
+
+import java.io.IOException;
+import java.util.List;
+import java.util.Objects;
+
+public class MergeExec extends LeafExec {
+
+    private final List<? extends PhysicalPlan> physSubPlans;
+    private final List<Attribute> output;
+
+    public MergeExec(Source source, List<? extends PhysicalPlan> physSubPlans, List<Attribute> output) {
+        super(source);
+        this.physSubPlans = physSubPlans;
+        this.output = output;
+    }
+
+    public List<LocalSupplier> suppliers() {
+        return physSubPlans.stream()
+            .filter(p -> LocalSourceExec.class.isAssignableFrom(p.getClass()))
+            .map(LocalSourceExec.class::cast)
+            .map(LocalSourceExec::supplier)
+            .toList();
```

Comment:
I think this should have some kind of comment.

I think it should also fail if it contains a non-LocalSourceExec rather than filter them out. Can we make the ctor only take those? Or do we end up transforming them in funny ways? It feels like they should be turned into a local-source-exec  by `EsqlSession`.

Reply:
Good idea. Done. [057381a](https://github.com/elastic/elasticsearch/pull/121948/commits/057381ae0a481b3f875e07671a43770579663edd). 

This could be simpler, but we eventually wanna get to a streamy mode of operations, at which point this will be removed.